File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        copy = x
        reverse = 0
        while copy != 0:
            reminder = copy % 10
            reverse = reverse * 10 + reminder
            copy //= 10
        return reverse == x

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int copy = x;
        long reverse = 0;
        while (copy != 0) {
            int rem = copy % 10;
            reverse = reverse * 10 + rem;
            copy /= 10;
        }
        return reverse == x;
    }
}